mos_tests/environment/os_actions.py

```diff
@@ -448,3 +451,72 @@ def execute_through_host(self, ssh, vm_host, cmd, creds=()):
         channel.close()
 
         return result
+
+    def wait_agents_alive(self, agt_ids_to_check):
+        logger.info('going to check if the agents alive')
+        helpers.wait(lambda: all([agt['alive'] for agt in
+                                  self.neutron.list_agents()['agents']
+                                  if agt['id'] in agt_ids_to_check]),
+                     timeout=5 * 60)
+
+    def add_net(self, router_id):
+        i =\
+            len(self.neutron.list_networks()['networks']) + 1
+        network =\
+            self.create_network(name='net%02d' % i)['network']
+        logger.info('network with id {} is created'.
+                    format(network['id']))
+        subnet = self.create_subnet(
+            network_id=network['id'],
+            name='net%02d__subnet' % i,
+            cidr="192.168.%d.0/24" % i)
+        logger.info('subnet with id {} is created'.
+                    format(subnet['subnet']['id']))
+        self.router_interface_add(
+            router_id=router_id,
+            subnet_id=subnet['subnet']['id'])
+        return network['id']
+
+    def add_server(self, network_id, key_name, hostname, sg_id):
+        i = len(self.nova.servers.list()) + 1
+        zone = self.nova.availability_zones.find(zoneName="nova")
+        self.create_server(
+            name='server%02d' % i,
+            availability_zone='{}:{}'.format(zone.zoneName, hostname),
+            key_name=key_name,
+            nics=[{'net-id': network_id}],
+            security_groups=[sg_id])
+
+    def reshedule_router_on_primary(self, router_id, primary_fqdn):
```

Comment:
In this case the easiest way  is to use host's fqdn, since the l3 agents contains them.
I would propose then to name this method reschedule_router_to_primary_by_fqdn